xmtp/src/lib.rs

```diff
@@ -37,9 +37,14 @@ pub trait Store<I> {
     fn store(&self, into: &mut I) -> Result<(), StorageError>;
 }
 
-// Fetches all instances of a model from the underlying data store
 pub trait Fetch<T> {
-    fn fetch(&mut self) -> Result<Vec<T>, StorageError>;
+    type Key<'a>;
+    // Fetches all instances of a model from the underlying data store
+    fn fetch_all(&mut self) -> Result<Vec<T>, StorageError>;
+
+    // Fetches a single instance by key of a model from the underlying data store
+    #[allow(clippy::needless_lifetimes)]
```

Comment:
Without the lifetime the `fetch_one` would be limited to non-reference types. By including the lifetime in the trait it can be elided in most implementations that don't require it. 

This is a clippy false positive from what I can tell. https://users.rust-lang.org/t/false-alert-from-clippy-lifetime-elision-for-str/54246